src/main.zig

```diff
@@ -878,11 +878,23 @@ fn getVersionUrl(
     app_data_path: []const u8,
     semantic_version: SemanticVersion,
 ) !DownloadUrl {
-    if (build_options.exe == .zls) return DownloadUrl.initOfficial(std.fmt.allocPrint(
-        arena,
-        "https://builds.zigtools.org/zls-{s}-{}.{s}",
-        .{ os_arch, semantic_version, archive_ext },
-    ) catch |e| oom(e));
+    if (build_options.exe == .zls) {
+        const info_url = try std.fmt.allocPrint(arena, "https://releases.zigtools.org/v1/zls/select-version?compatibility=only-runtime&zig_version={}", .{semantic_version});
+        const info_uri = try std.Uri.parse(info_url);
+        const info_basename = try std.fmt.allocPrint(arena, "download-index-zls-{}.json", .{semantic_version});
+        const index_path = try std.fs.path.join(arena, &.{ app_data_path, info_basename });
+
+        try fetchFile(arena, info_url, info_uri, index_path);
+        const index_content = blk: {
+            // since we just downloaded the file, this should always succeed now
+            const file = try std.fs.cwd().openFile(index_path, .{});
+            defer file.close();
+            break :blk try file.readToEndAlloc(arena, std.math.maxInt(usize));
+        };
+        defer arena.free(index_content);
+
+        return extractUrlFromZlsDownloadIndex(arena, index_path, index_content);
+    }
```

Comment:
If I'm reading this right, this appears to do the same thing as what we do in the mach case, download the index and get the actual download URL from there.  However, you're missing the index cache we do for the mach case.  Seems like we should re-use the same code for both cases.